### DISCUSSION TITLE
Intent 3 analysis

### DISCUSSION THREAD

**Discussion by Kevsy** [camaraproject/EdgeCloud#174] [General]
### Developer intents: Provisioning intents

| Intent 3 | 5SED |5MEE |EXRA|EXRC| EXCA | EXSM|  GMEC |
| -------- |------| ----|----| ---|------|-----|-------|
| 3. "I can discover the geographical regions covered by the operators MECs"   | No            |Yes     |Yes           |No     |Yes           |No    |No    |

| 5MEE           | Details       |Comments|
| -------------  |-------------| ------|
| `GET /regions` | List the geographical regions supported, and the associated zones| Additional method to `GetOptimalPlatformsByRegion`|

| EXRA           | Details       |Comments|
| -------------  |-------------| ------|
| `POST /api/v1/auth/ctrl/ShowCloudlet` | Show Cloudlets. Lists all the cloudlets managed from Edge Controller. Includes a region parameter per cloudlet | This method does not list directly all available regions, there does not seem to exist a direct method for that. |

**Comment by ThomasEdgeXR** [comment]
The API to obtain the regions supported by a controller is actually controller/show which is part of the contribution in https://github.com/camaraproject/EdgeCloud/blob/main/code/API_definitions/edgexr-nb-openapi.yaml.

**Comment by Kevsy** [comment]
Thanks Thomas - so which if the table cells above would say yes to reflect that - is it EXRC?

**Comment by gunjald** [comment]
This capability is not part of the Capgemini proposal i.e. GMEC

**Comment by maheshc01** [comment]
the table accurately reflects the support for the intent in 5GFF MEC Exposure & Experience Management API

**Comment by Kevsy** [comment]
Updated following feedback

**Comment by ThomasEXR** [comment]
This are the flow diagrams for corresponding EdgeXR API. Note that the notion of "region" is up to the deploying operator to define, i.e. it doesn't necessarily need to reflect geographical regions (but i can). 

show

```mermaid
sequenceDiagram
    participant developer
    participant operator    
    Note over developer,operator: PRE App determined proper Operator Gateway as per attached network
    Note over developer,operator: PRE App developer registered, <br/> authenticated and authorised
    developer->>operator: POST /Show
    operator->>developer: Iist of supported regions by that operator
```
